roles/edpm_nodes_validation/tasks/main.yml

```diff
@@ -63,11 +63,25 @@
     - edpm_nodes_validation_validate_controllers_icmp|bool
     - edpm_nodes_validation_ping_test_ips | length > 0
 
-- name: Verify the configured FQDN vs /etc/hosts
+- name: Verify a FQDN is set block
+  when:
+    - edpm_nodes_validation_check_for_fqdn|bool
+  block:
+  - name: Run hostname
+    command: hostname
+    register: hostname
+
+  - name: Verify a FQDN is set
```

Comment:
Setting fqdn in /etc/hostname is not mandatory. Can you elaborate what tls-e issue we're solving with this?

Reply:
TLS isn't the main concern. The fqdn MUST NOT change as it is the uid for many services.
If hostname is not explicitly the fqdn then AFAIK it can e.g if a DHCP lease could set the domain. Resolving this in the past required intrusive database surgery.

cloud-init sets it AFAICT (runs `hostname-ctl set-hostname <fqdn>`)

Reply:
> If hostname is not explicitly the fqdn then AFAIK it can e.g if a DHCP lease could set the domain.

I'm not sure I get the context here. It would be good to know when and how this could happen in the context of EDPM nodes (wrt dhcp lease) and what past issues we're referring to. We discussed the issues from services (ex.nova) pov in https://github.com/openstack-k8s-operators/edpm-ansible/pull/587 and decided not to change /etc/hostname.


> cloud-init sets it AFAICT (runs `hostname-ctl set-hostname <fqdn>`)

We use cloud-init with baremetal provisioning[1] and I don't see it setting fqdn as hostname[2].

[1] https://github.com/openstack-k8s-operators/openstack-baremetal-operator/blob/main/templates/openstackbaremetalset/cloudinit/userdata#L2-L4

[2] https://logserver.rdoproject.org/12/812/e4ca87621bb89d6ab5a56ab8a92195079e9eafbc/github-check/cifmw-crc-podified-edpm-baremetal/c531266/ci-framework-data/logs/192.168.122.100/etc/hostname